### ISSUE TITLE
load gemm_config error

### ISSUE THREAD

**Issue by clu2gt** [NVIDIA/FasterTransformer#365]
### Branch/Tag/Commit

v3.1

### Docker Image Version

no image docker

### GPU name

1060

### CUDA Driver

NVIDIA-SMI 470.141.03

### Reproduced Steps

```shell
I applied the encoder acceleration code to my model inference, and my model is not the standard bert model.
My model has  more embedding layers than the standard bert-base model, so after the embedding is computed, I use the dynamic acceleration library by Faster Transformers for the subsequent code. 
Although the inference works fine, the following warning appears for each inference：

[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][BertEncoderTransformer] Loading GEMM algorithms error, using default GEMM algorithms!
[WARNING][OpenMultiHeadAttention] Loading GEMM algorithms error, using default GEMM algorithms!
```

**Comment by clu2gt** [comment]
and that's my forward code
`  
  def forward(self, hidden_states, sequence_lengths):
        sequence_lengths = sequence_lengths.type(torch.int32)
        batch_length = torch.max(sequence_lengths).item()
        attention_mask = sequence_mask(sequence_lengths, batch_length, False).half()
        output_mask = sequence_mask(sequence_lengths, batch_length).to(attention_mask.dtype).unsqueeze(-1).half()
        if self.remove_padding:
            hidden_states, sequence_id_offset = self.build_mask_remove_padding(hidden_states, sequence_lengths)
            trt_seq_len = torch.cumsum(
                torch.cat([torch.tensor([0]).to(sequence_lengths).cuda(), sequence_lengths], dim=0), dim=0).to(
                torch.int32).cuda()
        else:
            sequence_id_offset = torch.tensor(0).to(torch.int32).cuda()
            batch = hidden_states.size(0)
            max_seq_len = hidden_states.size(1)
            padding_offset = torch.arange(0, batch * max_seq_len, max_seq_len).cuda()
            squence_offset_with_padding = sequence_lengths + padding_offset
            c = torch.cat([padding_offset, squence_offset_with_padding], dim=0)
            c_r = torch.reshape(c, [2, -1])
            t = torch.transpose(c_r, 0, 1)
            trt_seq_len = torch.reshape(t, [-1])
            trt_seq_len = torch.cat([trt_seq_len, torch.tensor([batch * max_seq_len]).to(trt_seq_len.dtype).cuda()],
                                    dim=0).to(torch.int32)
        for i in range(self.layer_num):
            hidden_states = self.encoders[i].forward(hidden_states, attention_mask, trt_seq_len, sequence_id_offset)
        if self.remove_padding:
            hidden_states = self.rebuild_padding(hidden_states, sequence_id_offset, attention_mask, 0)
        hidden_states = hidden_states * output_mask
        return (hidden_states,)`

**Comment by byshiue** [comment]
Please refer https://github.com/NVIDIA/FasterTransformer/blob/v3.1/docs/encoder_guide.md#encoder-process. You need to generate `gemm_config` before running the inference.

**Comment by clu2gt** [comment]
> Please refer https://github.com/NVIDIA/FasterTransformer/blob/v3.1/docs/encoder_guide.md#encoder-process. You need to generate `gemm_config` before running the inference.

I have generated this file, and put it in inference dir. but this warning is still thrown when the length of the input sentence does not match the seq length in the file.

**Comment by byshiue** [comment]
If you have several length, you need run the gemm test multiple times and concatenate all results into `gemm_config.in`. 
If there are too many cases, you can also use default gemm algo directly.

**Comment by clu2gt** [comment]
> If you have several length, you need run the gemm test multiple times and concatenate all results into `gemm_config.in`. If there are too many cases, you can also use default gemm algo directly.

there are too many cases， So how should I use the default gemm algo directly？

**Comment by byshiue** [comment]
> > If you have several length, you need run the gemm test multiple times and concatenate all results into `gemm_config.in`. If there are too many cases, you can also use default gemm algo directly.
> 
> there are too many cases， So how should I use the default gemm algo directly？

When the gemm wrapper cannot find the gemm config, it will use default algo by default. Like the warning message: `using default GEMM algorithms`.

**Comment by clu2gt** [comment]
> > > If you have several length, you need run the gemm test multiple times and concatenate all results into `gemm_config.in`. If there are too many cases, you can also use default gemm algo directly.
> > 
> > 
> > there are too many cases， So how should I use the default gemm algo directly？
> 
> When the gemm wrapper cannot find the gemm config, it will use default algo by default. Like the warning message: `using default GEMM algorithms`.

I want to use the default GEMM algorithms directly and not generate warnings, because the large number of such warnings in the logs affects the view of other error messages，Is there any way to solve this problem?

**Comment by byshiue** [comment]
Remove https://github.com/NVIDIA/FasterTransformer/blob/v3.1/fastertransformer/cuda/open_attention.h#L229 and https://github.com/NVIDIA/FasterTransformer/blob/v3.1/fastertransformer/bert_encoder_transformer.h#L395

**Comment by clu2gt** [comment]
OK，thank you！
